### PR TITLE
chore: refactor entry into deployment and runtime

### PR DESCRIPTION
**Pull request by Emyrk**
- RuntimeEntry has no startup value, and omits functions required to be serpent compatible.
- `DeploymentEntry` has no `New()` function because it is always instantiated from a `serpent` context. There is no point to implementing a `New()` function that will never be used in prod.
- Now there exists a `Manager` that will be instantiated once for the app. The `Manager` produces `Resolvers` such that we can implement some shared cache.
- Resolvers do not know about namespaces. Only managers (the factory) does now.

### CODE REVIEW

**Review by dannykopping — APPROVED** [review]
LGTM! :shipit:

File: coderd/runtimeconfig/resolver.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+// settings. Essentially, it reuses the shared "cache" that all resolvers should
+// use.
+type MemoryCachedResolver struct {
+	cache *syncmap.Map[string, cacheEntry]
+
+	wrapped Resolver
+}
+
+func NewMemoryCachedResolver(cache *syncmap.Map[string, cacheEntry], wrapped Resolver) *MemoryCachedResolver {
+	return &MemoryCachedResolver{
```
[the remark this answers is not in the document]
Comment (Emyrk) [reply coderd/runtimeconfig/resolver.go:103]: I think we should actually delete the cache resolver before we merge to `main`. Was just testing out some api usage with it.

File: coderd/runtimeconfig/entry_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+		var field runtimeconfig.DeploymentEntry[*serpent.String]
+		field.Initialize("my-field")
+		field.SetStartupValue(base.String())
+		// Check that default has been set.
+		require.Equal(t, base.String(), field.StartupValue().String())
+		// Validate that it returns that value.
+		require.Equal(t, base.String(), field.String())
+		// Validate that there is no org-level override right now.
+		_, err := field.Resolve(ctx, mgr.DeploymentResolver(db))
```
[the remark this answers is not in the document]
Comment (Emyrk) [reply coderd/runtimeconfig/entry_test.go:82]: :+1:

File: coderd/runtimeconfig/manager.go [hunk 3]
```diff
@@ -1,80 +1,50 @@
...
+	lastUpdated time.Time
 }
 
-func (n NoopManager) Scoped(string) Manager {
-	return n
+// MemoryCacheManager is an example of how a caching layer can be added to the
+// resolver from the manager.
+// TODO: Delete MemoryCacheManager and implement it properly in 'StoreManager'.
+type MemoryCacheManager struct {
+	cache *syncmap.Map[string, cacheEntry]
```
[the remark this answers is not in the document]
Comment (Emyrk) [reply coderd/runtimeconfig/manager.go:35]: Handy little guy

**Pull request merged**